src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -881,6 +889,311 @@ public void RestoreOriginalParentTest()
+
+        [InlineData($"Unknown,          W3C,                Unknown,            null,       true,                   true,               W3C,                    Default")]
+        [InlineData($"W3C,              Unknown,            Unknown,            null,       true,                   true,               W3C,                    W3C")]
+        [InlineData($"Unknown,          Unknown,            W3C,                null,       true,                   true,               W3C,                    W3C")]
+        [InlineData($"Unknown,          Unknown,            Unknown,            W3C,        true,                   true,               W3C,                    W3C")]
+        [InlineData($"Unknown,          Unknown,            Unknown,            null,       false,                  true,               W3C,                    W3C")]
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestIdFormats(string data)
+        {
+            RemoteExecutor.Invoke((d) => {
+
+                //
+                // Extract the data
+                //
+                string [] parts = d.Split(',');
+
+                ActivityIdFormat defaultId = (ActivityIdFormat) Enum.Parse(typeof(ActivityIdFormat), parts[0].Trim());
+                ActivityIdFormat idToForce = (ActivityIdFormat) Enum.Parse(typeof(ActivityIdFormat), parts[1].Trim());
+                ActivityIdFormat parentFormat = (ActivityIdFormat) Enum.Parse(typeof(ActivityIdFormat), parts[2].Trim());
+                string parentId = parts[3].Trim();
+                if (parentId == "null")
+                {
+                    parentId = null;
+                }
+                else if (parentId == "W3C")
+                {
+                    parentId = "00-99d43cb30a4cdb4fbeee3a19c29201b0-e82825765f051b47-01";
+                }
+
+                bool defaultParentContext = bool.Parse(parts[4].Trim());
+                bool checkTraceId         = bool.Parse(parts[5].Trim());
+
+                ActivityIdFormat expectedCreateFormat = parts[6].Trim() == "Default" ? Activity.DefaultIdFormat : (ActivityIdFormat) Enum.Parse(typeof(ActivityIdFormat), parts[6].Trim());
+                ActivityIdFormat expectedStartFormat  = parts[7].Trim() == "Default" ? Activity.DefaultIdFormat : (ActivityIdFormat) Enum.Parse(typeof(ActivityIdFormat), parts[7].Trim());
+
+                using ActivitySource aSource = new ActivitySource("FormatIdSource");
+                ActivityTraceId traceId = default;
+
+                //
+                //  Listener Creation
+                //
+
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource);
+                listener.SampleUsingParentId = (ref ActivityCreationOptions<string> activityOptions) =>
```

Comment:
Lets add a 2nd listener that only implements the Sample callback. This will let us confirm that Sample is always called for W3C formatted activities and never called for Hierarchal ones.